File: benchmarks/go/factorial.go
package main

import "fmt"

func factorial(n int64) int64 {
	if n <= 1 {
		return 1
	}
	return n * factorial(n-1)
}

func runLoop(n int64, acc int64) int64 {
	for n > 0 {
		acc += factorial(20)
		n--
	}
	return acc
}

func main() {
	fmt.Println(runLoop(10000000, 0))
}

File: benchmarks/rust/factorial.rs
fn factorial(n: i64) -> i64 {
    if n <= 1 { 1 } else { n.wrapping_mul(factorial(n - 1)) }
}

fn main() {
    let mut acc: i64 = 0;
    for _ in 0..10_000_000 {
        acc = acc.wrapping_add(factorial(20));
    }
    println!("{}", acc);
}
